pkg/target/target.go

```diff
@@ -136,12 +136,26 @@ func (h *K8sValidationTarget) HandleViolation(result *types.Result) error {
 	}
 
 	objMap, found, err := unstructured.NestedMap(rmap, "object")
-	if err != nil {
-		return errors.Wrap(err, "HandleViolation:NestedMap")
-	}
-	if !found {
-		return errors.New("no object returned in review")
+	if err != nil || !found {
+		if val, _, err2 := unstructured.NestedFieldNoCopy(rmap, "object"); val == nil && err2 == nil {
+			objMap2, found2, err3 := unstructured.NestedMap(rmap, "oldObject")
+			if err3 != nil {
+				return errors.Wrap(err, "HandleViolation:NestedMapOldObj")
+			}
+			if !found2 {
+				return errors.New("no object or oldObject returned in review")
+			}
+			objMap = objMap2
+		} else {
```

Comment:
This else seems incorrect

Reply:
It's certainly ugly, but I think correct.

Basically:

1. If `object` successfully retrieved continue `(err != nil || !found) == false => skip if statement`
2. else if `object` unsuccessfully retrieved, check if it's missing
   a. Try to get the value without assuming it's a map, which will return nil if the resource is defined (NestedMap raises an error if `value == nil`)
   b. if no errors and the value is nil, then either the previous error was due to `object == nil` or `found == false`, in either case, fall back to using `oldObject`
   c. if value is non-nil, then we have something unexpected in our request object, return the original error
   d. if error is non-nil, then there is a legitimate error retrieving `object`, return the original error

the `else` implements 2c and 2d

It's nasty because there are three return values, all of which are important in different ways for different reasons. I'm open to suggestions on how this could be made clearer.

Reply:
I could maybe just use unstructured.NestedFieldCopy() from the outset, which might streamline the error checking a bit at the cost of adding a cast to map[string]interface{}

Reply:
But then I'd also need to add error handling for that, so it may increase if-statement nesting.

Reply:
I guess `!found` is superfluous.

Anyway, added a wrapper function around nestedMap that makes the behavior align with our logic a little better, that should simplify things.